### PR TITLE
Deprecation cleanup

### PR DESCRIPTION
**Pull request by cleptric**
Update more deprecated method calls.

### CODE REVIEW

File: src/Controller/Component/PaginatorComponent.php [hunk 1]
```diff
@@ -282,7 +282,7 @@ protected function _setPagingParams()
      */
     public function config($key = null, $value = null, $merge = true)
     {
-        $return = $this->_paginator->config($key, $value, $merge);
+        $return = $this->_paginator->setConfig($key, $value, $merge);
```
Comment (chinpei215) [inline src/Controller/Component/PaginatorComponent.php]: This change is incorrect, as `config()` is really used as a combined setter/getter here. See the following code in the method. If we want to remove this call, probably the fix will be like this:
```php
public function config($key = null, $value = null, $merge = true)
{
    if (is_array($key) || func_num_args() >= 2) {
        $this->_paginator->setConfig($key, $value, $merge);

        return $this;
    }

    return $this->_paginator->getConfig($key);
}
```
Though I touched this method a little bit in my PR (#10959), feel free to fix it. I will rebase my commit then.

File: src/ORM/ResultSet.php [hunk 2]
```diff
@@ -185,7 +185,6 @@ public function __construct($query, $statement)
         $this->_useBuffering = $query->isBufferedResultsEnabled();
         $this->_defaultAlias = $this->_defaultTable->getAlias();
         $this->_calculateColumnMap($query);
-        $this->_calculateTypeMap();
```
Comment (markstory) [inline src/ORM/ResultSet.php]: Why is this being removed?

Comment (cleptric) [inline src/ORM/ResultSet.php]:
```
    /**
     * Creates a map of Type converter classes for each of the columns that should
     * be fetched by this object.
     *
     * @deprecated 3.2.0 Not used anymore. Type casting is done at the statement level
     * @return void
     */
    protected function _calculateTypeMap()
    {
    }
```

File: src/Controller/Component/PaginatorComponent.php (hunk 1 above)
Comment (cleptric) [inline src/Controller/Component/PaginatorComponent.php]: You're absolutely right. Thanks 👍 I'll revert it for the mean time.

File: src/ORM/ResultSet.php (hunk 2 above)
Comment (markstory) [inline src/ORM/ResultSet.php]: Ok.

**Comment by markstory** [comment]
Nice work!

**Pull request merged**